runtime/lua/vim/lsp/buf.lua

```diff
@@ -827,11 +827,15 @@ local function on_code_action_results(results, opts)
     return
   end
 
-  ---@param item {action: lsp.Command|lsp.CodeAction}
+  ---@param item {action: lsp.Command|lsp.CodeAction, ctx : lsp.HandlerContext}
   local function format_item(item)
-    local title = item.action.title:gsub('\r\n', '\\r\\n')
-    return title:gsub('\n', '\\n')
+    local source = vim.lsp.get_client_by_id(item.ctx.client_id).name or 'unknown'
+    return string
```

Comment:
arguably redundant if there is only 1 client attached, can we make it conditional?

```lua
local clients = vim.lsp.get_clients({bufnr=ctx.bufnr})
if #clients > 1 then
  ...
```